packages/framework/tests/Unit/Model/Order/Processing/OrderProcessorMiddleware/AddPaymentMiddlewareTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrameworkBundle\Unit\Model\Order\Processing\OrderProcessorMiddleware;
+
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Model\Order\Item\OrderItemTypeEnum;
+use Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessorMiddleware\AddPaymentMiddleware;
+use Shopsys\FrameworkBundle\Model\Payment\Payment;
+use Shopsys\FrameworkBundle\Model\Payment\PaymentData;
+use Shopsys\FrameworkBundle\Model\Payment\PaymentPriceCalculation;
+use Shopsys\FrameworkBundle\Model\Pricing\Price;
+use Tests\FrameworkBundle\Test\IsPriceEqual;
+use Tests\FrameworkBundle\Test\MiddlewareTestCase;
+
+class AddPaymentMiddlewareTest extends MiddlewareTestCase
```

Comment:
I wanted to have them in the framework, so the original behavior is properly tested and changes on the project do not cause tests to fail.

The order process is thoroughly tested already in the project-base and in case of custom changes, I believe the tests should be written independently.